src/index.ts

```diff
@@ -0,0 +1,356 @@
+  }
+
+  log(levelName: string, msg: string, metadata: Metadata|{},
+      callback: (err: Error, apiResponse: {}) => void) {
+    if (is.fn(metadata)) {
+      callback = metadata as (err: Error, apiResponse: {}) => void;
+      metadata = {};
+    }
+
+    if (this.levels[levelName] === undefined) {
+      throw new Error('Unknown log level: ' + levelName);
+    }
+
+    const levelCode = this.levels[levelName];
+    const stackdriverLevel = STACKDRIVER_LOGGING_LEVEL_CODE_TO_NAME[levelCode];
+
+    const entryMetadata: StackdriverEntryMetadata = {
+      resource: this.resource,
+    };
+
+    const data: StackdriverData = {};
+
+    // Stackdriver Logs Viewer picks up the summary line from the `message`
+    // property of the jsonPayload.
+    // https://cloud.google.com/logging/docs/view/logs_viewer_v2#expanding.
+    //
+    // For error messages at severity 'error' and higher, Stackdriver
+    // Error Reporting will pick up error messages if the full stack trace is
+    // included in the textPayload or the message property of the jsonPayload.
+    // https://cloud.google.com/error-reporting/docs/formatting-error-messages
+    // We prefer to format messages as jsonPayload (by putting it as a message
+    // property on an object) as that works is accepted by Error Reporting in
+    // for more resource types.
+    //
+    if (metadata && (metadata as Metadata).stack) {
+      msg += (msg ? ' ' : '') + (metadata as Metadata).stack;
+      data.serviceContext = this.serviceContext;
+    }
+    data.message = msg;
+
+    if (is.object(metadata)) {
+      data.metadata =
+          this.inspectMetadata ? mapValues(metadata, util.inspect) : metadata;
+
+      // If the metadata contains a httpRequest property, promote it to the
+      // entry metadata. This allows Stackdriver to use request log formatting.
+      // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
+      // Note that the httpRequest field must properly validate as HttpRequest
+      // proto message, or the log entry would be rejected by the API. We no do
+      // validation here.
+      if ((metadata as Metadata).httpRequest) {
+        entryMetadata.httpRequest = (metadata as Metadata).httpRequest;
+        delete (data.metadata as Metadata).httpRequest;
+      }
+    }
+
+    // tslint:disable-next-line:no-any
+    if (metadata && (metadata as any)[LOGGING_TRACE_KEY]) {
+      // tslint:disable-next-line:no-any
+      entryMetadata.trace = (metadata as any)[LOGGING_TRACE_KEY];
+      // tslint:disable-next-line:no-any
+      delete (data.metadata as any)[LOGGING_TRACE_KEY];
+    } else {
+      const trace = getCurrentTraceFromAgent();
+      if (trace) {
+        entryMetadata.trace = trace;
+      }
+    }
+
+    const entry = this.stackdriverLog.entry(entryMetadata, data);
+    // tslint:disable-next-line:no-any
+    (this.stackdriverLog as any)[stackdriverLevel](entry, callback);
+  }
+}
+
+// tslint:disable-next-line:no-any
```

Comment:
If you must disable a lint rule, you should include an explanation for why it is necessary.

Reply:
Done.